workers/unity/Packages/com.improbable.gdk.transformsynchronization/MonoBehaviours/TransformSynchronization.cs

```diff
@@ -0,0 +1,191 @@
+                    $" on should only be added to a GameObject linked to a SpatialOS entity");
+            }
+
+            // Check if this is actually needed - to not use the entity manager while in a system
+            StartCoroutine(ApplyStrategies());
+        }
+
+        private void OnDisable()
+        {
+            StopAllCoroutines();
+        }
+
+        private IEnumerator ApplyStrategies()
+        {
+            yield return null;
+            if (SetKinematicWhenNotAuthoritative)
+            {
+                spatialOSComponent.World.GetOrCreateManager<EntityManager>().AddComponent(spatialOSComponent.Entity,
+                    typeof(ManageKinematicOnAuthorityChangeTag));
+            }
+
+            ApplyReceiveStrategies();
+            ApplySendStrategies();
+        }
+
+        private void ApplyReceiveStrategies()
+        {
+            AddCommonReceiveComponents();
+
+            foreach (var strategy in ReceiveStrategies)
+            {
+                if (strategy.WorkerType != spatialOSComponent.Worker.WorkerType)
+                {
+                    continue;
+                }
+
+                strategy.Apply(spatialOSComponent.Entity, spatialOSComponent.World);
+            }
+        }
+
+        private void ApplySendStrategies()
+        {
+            if (SendStrategies == null || SendStrategies.Count == 0)
+            {
+                return;
+            }
+
+            AddCommonSendComponents();
+
+            foreach (var strategy in SendStrategies)
+            {
+                if (strategy.WorkerType != spatialOSComponent.Worker.WorkerType)
+                {
+                    continue;
+                }
+
+                strategy.Apply(spatialOSComponent.Entity, spatialOSComponent.World);
+            }
+        }
+
+        private void AddCommonReceiveComponents()
+        {
+            var entityManager = spatialOSComponent.World.GetOrCreateManager<EntityManager>();
+            entityManager.AddComponent(spatialOSComponent.Entity, typeof(SetTransformToGameObjectTag));
+
+            var transformComponent = transformReader.Data;
+
+            var defaultToSet = new TransformToSet
+            {
+                Position = transformComponent.Location.ToUnityVector3() + spatialOSComponent.Worker.Origin,
+                Velocity = transformComponent.Velocity.ToUnityVector3(),
+                Orientation = transformComponent.Rotation.ToUnityQuaternion(),
+                ApproximateRemoteTick = 0
+            };
+
+            var previousTransform = new DeferredUpdateTransform
+            {
+                Transform = transformComponent
+            };
+
+
+            entityManager.AddComponentData(spatialOSComponent.Entity, defaultToSet);
+            entityManager.AddComponentData(spatialOSComponent.Entity, previousTransform);
+            entityManager.AddBuffer<BufferedTransform>(spatialOSComponent.Entity);
+        }
+
+        private void AddCommonSendComponents()
```

Comment:
Seems right now it will make no difference but it does make conceptual sense. Will use a buffer now and we can review for optimisation later.

Reply:
Actually this is going into the main branch rather than here. This is just the - the api is as documented branch, so I'm not going to add large amounts of piping here. But elsewhere this change can be made. Will be pushed next week probably.

Reply:
Changed my mind again, will just throw it in.